addons/crm/models/crm_lead.py

```diff
@@ -1303,12 +1303,26 @@ def _merge_opportunity(self, user_id=False, team_id=False, auto_unlink=True, max
         return opportunities_head
 
     def _merge_get_fields_specific(self):
-        return {
+        fields_specific = {
             'description': lambda fname, leads: '<br/><br/>'.join(desc for desc in leads.mapped('description') if not is_html_empty(desc)),
             'type': lambda fname, leads: 'opportunity' if any(lead.type == 'opportunity' for lead in leads) else 'lead',
             'priority': lambda fname, leads: max(leads.mapped('priority')) if leads else False,
+            'tag_ids': lambda fname, leads: leads.mapped('tag_ids'),
+            'lost_reason': lambda fname, leads:
+                False if leads and leads[0].probability
+                else next((lead.lost_reason for lead in leads if lead.lost_reason), False),
         }
 
+        # Address related fields, all fields are propagated or none
+        fields_specific.update({
+            field: lambda fname, leads: next((
+                lead[fname] for lead in leads
+                if any(lead[field] for field in PARTNER_ADDRESS_FIELDS_TO_SYNC)), False)
```

Comment:
Actually it is a bit tricky, thinking of it. Do we want to accept an address from an higher ranker lead, or a full address from a lower ranked lead ? Would day that we could fetch address information of all merged leads, and take the one with the most information (taking higher ranked lead if equivalent number of fields given).